45-topi-antelope/technical_specification.md

```diff
@@ -0,0 +1,83 @@
+# Structured Logging Tools (Topi Antelope)
+**Epic Type:** Implementation Epic
+
+Epic planning and implementation follow the
+[Epic Planning and Marathon SOP](https://docs.ghga-dev.de/main/sops/sop001_epic_planning.html).
+
+**Attention: Please do not put any confidential content here.**
+
+## Scope
+### Outline:
+The aim of this epic is to implement tools that enable consistent structured logging,
+with the result being a set of minimally-invasive, configurable logging tools. The
+logging functionality is not meant to live within providers or protocols within hexkit,
+but rather be provided as a utility similar to the basic correlation ID functionality.
+In addition, `uvicorn` needs to be configured such that its formatting is consistent
+with other logs.
+
+### Included/Required:
+- Logging Tools in Hexkit
+- Consolidating Uvicorn Logging
+- Pilot Repository
+
+### Not Included:
+- Updating services to use new logging tools.
+
+
+## Additional Implementation Details:
+
+### Logging Tools in `Hexkit`
+There needs to be a way to easily invoke a logger object with minimal boilerplate code.
+There should also be a dedicated logging configuration class that subclasses `BaseSettings`.
+
+The configuration should at least include the log level for the service.
+When employing loggers in a service, it should be as easy as calling a function with a logger
+name as an argument and then using the object to log as needed.
```

Comment:
I don't understand what is meant by this but happy to wait for the code :)

Reply:
However, I would avoid adding logging details to function signatures if this is what you have meant. Moreover, I currently don't see a need for multiple loggers, or do we need it?

Reply:
No, what I was referring to was instantiating a logger at the top of a module in order to use it throughout the module as needed. Usually you would call `logging.getLogger()` (maybe with `__name__` or something as an arg), then do the basic config before using it. What I meant here was that our logger(s) should be just as simple to use.